package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -726,6 +777,635 @@ def __setstate__(self, state):
+    ----
+    Parellel writing with the use of a MPI communicator and the ``'mpio'``
+    HDF5 driver is currently not supported.
+
+
+    Raises
+    ------
+    RuntimeError
+        when `H5PY`_ is not installed
+    ValueError
+        when ``n_atoms`` is 0
+    ValueError
+        when the file is opened with ``driver=='mpio'`` or
+        with an MPI communicator
+    ValueError
+        when ``chunks=False`` but the user did not specify ``n_frames``
+    ValueError
+        when ``positions``, ``velocities``, and ``forces`` are all
+        set to ``False``
+    TypeError
+        when the input object is not a :class:`Universe` or
+        :class:`AtomGroup`
+    IOError
+        when ``n_atoms`` of the :class:`Universe` or :class:`AtomGroup`
+        being written does not match ``n_atoms`` passed as an argument
+        to the writer
+    ValueError
+        when any of the optional ``timeunit``, ``lengthunit``,
+        ``velocityunit``, or ``forceunit`` keyword arguments are
+        unrecognizable by MDAnalysis
+
+
+    .. versionadded:: 2.0.0
+
+    """
+
+    format = 'H5MD'
+    multiframe = True
+    #: These variables are not written from :attr:`Timestep.data`
+    #: dictionary to the observables group in the H5MD file
+    data_blacklist = ['step', 'time', 'dt']
+
+    #: currently written version of the file format
+    H5MD_VERSION = (1, 1)
+
+    # This dictionary is used to translate MDAnalysis units to H5MD units.
+    # (https://nongnu.org/h5md/modules/units.html)
+    _unit_translation_dict = {
+        'time': {
+            'ps': 'ps',
+            'fs': 'fs',
+            'ns': 'ns',
+            'second': 's',
+            'sec': 's',
+            's': 's',
+            'AKMA': 'AKMA'},
+        'length': {
+            'Angstrom': 'Angstrom',
+            'angstrom': 'Angstrom',
+            'A': 'Angstrom',
+            'nm': 'nm',
+            'pm': 'pm',
+            'fm': 'fm'},
+        'velocity': {
+            'Angstrom/ps': 'Angstrom ps-1',
+            'A/ps': 'Angstrom ps-1',
+            'Angstrom/fs': 'Angstrom fs-1',
+            'A/fs': 'Angstrom fs-1',
+            'Angstrom/AKMA': 'Angstrom AKMA-1',
+            'A/AKMA': 'Angstrom AKMA-1',
+            'nm/ps': 'nm ps-1',
+            'nm/ns': 'nm ns-1',
+            'pm/ps': 'pm ps-1',
+            'm/s': 'm s-1'},
+        'force':  {
+            'kJ/(mol*Angstrom)': 'kJ mol-1 Angstrom-1',
+            'kJ/(mol*nm)': 'kJ mol-1 nm-1',
+            'Newton': 'Newton',
+            'N': 'N',
+            'J/m': 'J m-1',
+            'kcal/(mol*Angstrom)': 'kcal mol-1 Angstrom-1',
+            'kcal/(mol*A)': 'kcal mol-1 Angstrom-1'}}
+
+    def __init__(self,
+                 filename,
+                 n_atoms,
+                 driver=None,
+                 comm=None,
+                 convert_units=True,
+                 chunks=None,
+                 compression=None,
+                 compression_opts=None,
+                 **kwargs):
```

Comment:
Make all documented kwargs explicit instead of `**kwargs`. Just leave `**kwargs` to catch any unprocessed kwargs.

This will help a lot with the documentation and seeing defaults.